toctoc/core/src/test/scala/io.buildo.toctoc/core/authentication/TokenBasedRecoveryFlowSpec.scala

```diff
@@ -0,0 +1,73 @@
+package io.buildo.toctoc
+package core
+package authentication
+
+import org.scalatest._
+import org.scalatest.prop._
+import org.scalactic.TypeCheckedTripleEquals
+import org.scalacheck.magnolia._
+import cats.effect.IO
+import cats.data.EitherT
+import java.time.Duration
+
+final class TokenBasedRecoveryFlowSpec
+    extends PropSpec
+    with PropertyChecks
+    with TypeCheckedTripleEquals {
+
+  import TokenBasedAuthentication._
+  import TokenBasedRecovery._
+
+  val loginDomain: LoginDomain[IO] =
+    InMemoryAuthenticationDomain.create(Map.empty)
+
+  val tokenDomain: AccessTokenDomain[IO] =
+    InMemoryAuthenticationDomain.create(Map.empty)
+
+  val recoveryFlow = TokenBasedRecoveryFlow.create(loginDomain, tokenDomain, Duration.ofDays(1))
+
+  def check[A](test: => IO[Either[A, Assertion]]): Assertion =
+    test
+      .unsafeRunSync()
+      .fold(
+        error => fail(error.toString()),
+        assertion => assertion,
+      )
+
+  property("recovery works for existing users") {
+    forAll { (s: UserSubject, username: String, password: String) =>
+      check {
+        (for {
+          registerResult <- EitherT(recoveryFlow.registerForRecovery(s))
+          (f, token) = registerResult
+          f2 <- EitherT(f.recoverLogin(token, password) { _ =>
+            IO(Some(username))
+          })
+          authenticateResult <- EitherT(f2.loginD.authenticate(Login(username, password)))
+          (_, subject) = authenticateResult
```

Comment:
super minor, but I kinda hate that we cannot do:

```scala
(_, subject) <- EitherT(...)
```

This is solved by https://github.com/oleg-py/better-monadic-for which we might want to introduce in our build (I've used in on personal projects and it works well!)